UBL additional cleanup

thinkyhead opened the pull request:
Some changes by @Bob-the-Kuhn from #6086, plus additional fixes.

- Various patches to formatting, trailing whitespace, spelling, etc.
- Convert data and methods in `class unified_bed_leveling` to `static` to reduce code size and calling overhead.
- Make `G26` compatible with workspace offsets.
- Disable `G26` by default

[Concise diff](https://github.com/MarlinFirmware/Marlin/pull/6174/files?w=1)

Bob-the-Kuhn committed:
Reduce UBL RAM usage by making G26/G29 optional

thinkyhead committed:
Further repairs to UBL, comments, spacing

thinkyhead committed:
Disable UBL_G26_MESH_EDITING by default

Roxy-3D commented:
> On the agenda next: Make all data and methods in the unified_bed_leveling class static to reduce code size and overhead.

I don't understand why this would reduce code size.   Can you explain why this would be so?

thinkyhead committed:
More data in UBL class, make it a static class

- Make all `unified_bed_leveling` data/methods static
- Move some UBL-related variables into the class
- Replace `map_[xy]_index_to_bed_location` with `mesh_index_to_[xy]pos`

thinkyhead committed:
Make G26 compatible with G92 and M206

thinkyhead commented:
> I don't understand why this would reduce code size. Can you explain why this would be so?

- Non-static member functions include a hidden `this` value passed as the first argument.
- Non-static data members are accessed by applying an offset to `this` or the given pointer (`ubl.`)

thinkyhead commented:
This is the last round of changes to get the UBL code into general compliance. Next we'll bring the LCD interface code into compliance.

thinkyhead commented:
This is the last round of changes to get the UBL code into general compliance. Next we'll bring the LCD interface code into compliance.

Note that the `ubl.state` data structure has been altered to be incompatible, so you'll need to reset, regenerate, and re-save your UBL eeprom data. I don't see any versioning or checksum on the UBL data. Probably that should be added.

Roxy-3D commented:
>> I don't understand why this would reduce code size. Can you explain why this would be so?
> - Non-static member functions include a hidden this value passed as the first argument.
> - Non-static data members are accessed by applying an offset to this

OK...   So if this is true...  That would imply that declaring a class's data 'static' means there is really only one instance of the data.  It is almost like a global variable.   And the only reason we can get away with doing this is because we only have one instance of the class instantiated.   Right?

thinkyhead commented:
Correct. It's only appropriate with so-called "singletons."
